src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,148 @@
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import cn from 'classnames';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+
+import { Loader } from '../Loader';
+import { ErrorLoadPage } from '../ErrorLoadPage';
+import { PersoneLink } from '../PersoneLink';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const { slug } = useParams();
+
+  const getDataFromServer = async () => {
+    setIsLoading(true);
+    setIsError(false);
+
+    try {
+      const peopleFromServer = await getPeople();
+
+      const personWithParents = peopleFromServer
+        .map(person => {
+          const mother = peopleFromServer
+            .find(parent => parent.name === person.motherName);
+          const father = peopleFromServer
+            .find(parent => parent.name === person.fatherName);
+
+          return {
+            ...person,
+            mother,
+            father,
+          };
+        });
+
+      setPeople(personWithParents);
+    } catch (error) {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getDataFromServer();
+  }, []);
+
+  const content = () => {
+    if (isLoading) {
+      return <Loader />;
+    }
+
+    if (isError) {
+      return <ErrorLoadPage />;
+    }
+
+    if (!people.length) {
+      return (
+        <p data-cy="noPeopleMessage">
+          There are no people on the server
+        </p>
+      );
+    }
```

Comment:
Also maybe you need to check the condition for that error. 
Now when I open people page it shows this error just for a moment, before the loader appears. 
I think happens because when the component is mounted you initialize people with empty array.

![image](https://github.com/mate-academy/react_people-table-basics/assets/88306382/c05d8dd1-430d-432a-8100-17a42b4e83d0)

Reply:
Thanks a lot!